website/content/docs/connect/gateways/ingress-gateway/tls-external-service.mdx

```diff
@@ -0,0 +1,245 @@
+
+## Overview
+
+The following process describes the general procedure for configuring ingress gateways to serve TLS certificates sourced from external services:
+
+1. Configure static SDS clusters in the ingress gateway service definition.
+1. Register the service definition.
+1. Configure TLS client authentication
+1. Start Envoy.
+1. Configure SDS settings in an ingress gateway configuration entry.
+1. Register the ingress gateway configuration entry with Consul.  
+
+## Requirements
+
+- The external service must implement Envoy's [gRPC secret discovery service (SDS) API](https://www.envoyproxy.io/docs/envoy/latest/configuration/security/secret).
+- You should have some familiarity with Envoy configuration and the SDS protocol.
+- The [`connect.enabled` parameter](/consul/docs/agent/config/config-files#connect) must be set to `true` for all server agents in the Consul datacenter.
+- The [`ports.grpc` parameter](/consul/docs/agent/config/config-files#connect) must be configured for all server agents in the Consul datacenter.
+
+### ACL requirements
+
+If ACLs are enabled, you must present a token when registering ingress gateways that grant the following permissions:
+
+- `service:write` for the ingress gateway's service name 
+- `service:read` for all services in the ingress gateway's configuration entry
+- `node:read` for all nodes of the services in the ingress gateway's configuration entry.
+
+These privileges authorize the token to route communications to other services in the mesh. If the Consul client agent on the gateway's node is not configured to use the default gRPC port, `8502`, then the gateway's token must also provide `agent:read` for its node's name in order to discover the agent's gRPC port. gRPC is used to expose Envoy's xDS API to Envoy proxies.
+
+## Configure static SDS clusters
+
+You must define one or more additional static clusters in the ingress gateway service definition for each Envoy proxy associated with the gateway. The additional clusters define how Envoy should connect to the required SDS services.
+
+Configure the static clusters in the [`Proxy.Config.envoy_envoy_extra_static_clusters_json`](/consul/docs/connect/proxies/envoy#envoy_extra_static_clusters_json) parameter in the service definition. 
+
+The clusters must provide connection information and any necessary authentication information, such as mTLS credentials.
+
+You must manually register the ingress gateway with Consul proxy to define extra clusters in Envoy's bootstrap configuration. You can not use the `-register` flag with `consul connect envoy -gateway=ingress` to automatically register the proxy to define static clusters. 
+
+In the following example, the `public-ingress` gateway includes a static cluster named `sds-cluster` that specifies paths to the SDS certificate and SDS certification validation files:
+
+
+<CodeBlock heading="public-ingress-service.hcl">
```

Comment:
```suggestion
<CodeBlockConfig heading="public-ingress-service.hcl">
```

Reply:
there's a couple of these, so that's why i didn't do suggestions for all of them 😛